packages/platforms/delta/x86-64/x86-64-delta-ag9032v1/onlp/builds/src/module/src/fani.c

```diff
@@ -0,0 +1,422 @@
+    { "3-002c/fan3_fault", "3-002c/fan3_input", "3-002c/fan3_input_percentage" },
+    { "3-002c/fan4_fault", "3-002c/fan4_input", "3-002c/fan4_input_percentage" },
+    { "3-002c/fan5_fault", "3-002c/fan5_input", "3-002c/fan5_input_percentage" },
+    { "3-002d/fan1_fault", "3-002d/fan1_input", "3-002d/fan1_input_percentage" },
+    { "3-002d/fan2_fault", "3-002d/fan2_input", "3-002d/fan2_input_percentage" },
+    { "3-002d/fan3_fault", "3-002d/fan3_input", "3-002d/fan3_input_percentage" },
+    { "3-002d/fan4_fault", "3-002d/fan4_input", "3-002d/fan4_input_percentage" },
+    { "3-002d/fan5_fault", "3-002d/fan5_input", "3-002d/fan5_input_percentage" },
+    { "4-0058/psu_fan1_fault", "4-0058/psu_fan1_speed_rpm", "4-0058/psu_fan1_duty_cycle_percentage" },
+    { "4-0058/psu_fan1_fault", "4-0058/psu_fan1_speed_rpm", "4-0058/psu_fan1_duty_cycle_percentage" }
+};
+
+#define MAKE_FAN_INFO_NODE_ON_FAN_BOARD(id) \
+    { \
+        { ONLP_FAN_ID_CREATE(FAN_##id##_ON_FAN_BOARD), "Chassis Fan "#id, 0 }, \
+        0x0, \
+        (ONLP_FAN_CAPS_SET_RPM | ONLP_FAN_CAPS_GET_RPM), \
+        0, \
+        0, \
+        ONLP_FAN_MODE_INVALID, \
+    }
+
+#define MAKE_FAN_INFO_NODE_ON_PSU(psu_id, fan_id) \
+    { \
+        { ONLP_FAN_ID_CREATE(FAN_##fan_id##_ON_PSU##psu_id), "Chassis PSU-"#psu_id " Fan "#fan_id, 0 }, \
+        0x0, \
+        (ONLP_FAN_CAPS_SET_PERCENTAGE | ONLP_FAN_CAPS_GET_RPM | ONLP_FAN_CAPS_GET_PERCENTAGE), \
+        0, \
+        0, \
+        ONLP_FAN_MODE_INVALID, \
+    }
+
+/* Static fan information */
+onlp_fan_info_t linfo[] = {
+    { }, /* Not used */
+    MAKE_FAN_INFO_NODE_ON_FAN_BOARD(1),
+    MAKE_FAN_INFO_NODE_ON_FAN_BOARD(2),
+    MAKE_FAN_INFO_NODE_ON_FAN_BOARD(3),
+    MAKE_FAN_INFO_NODE_ON_FAN_BOARD(4),
+    MAKE_FAN_INFO_NODE_ON_FAN_BOARD(5),
+    MAKE_FAN_INFO_NODE_ON_FAN_BOARD(6),
+    MAKE_FAN_INFO_NODE_ON_FAN_BOARD(7),
+    MAKE_FAN_INFO_NODE_ON_FAN_BOARD(8),
+    MAKE_FAN_INFO_NODE_ON_FAN_BOARD(9),
+    MAKE_FAN_INFO_NODE_ON_FAN_BOARD(10),
+    MAKE_FAN_INFO_NODE_ON_PSU(1,1),
+    MAKE_FAN_INFO_NODE_ON_PSU(2,1),
+};
+
+#define VALIDATE(_id)                           \
+    do {                                        \
+        if(!ONLP_OID_IS_FAN(_id)) {             \
+            return ONLP_STATUS_E_INVALID;       \
+        }                                       \
+    } while(0)
+
+static int
+dni_fani_info_get_fan(int local_id, onlp_fan_info_t* info)
+{
+    int rpm = 0;
+    char fullpath[100] = {0};
+    uint8_t present_bit=0x00, bit=0x00;
+
+    mux_info_t mux_info;
+    mux_info.offset = SWPLD_PSU_FAN_I2C_MUX_REG;
+    mux_info.channel = 0x05;
+    mux_info.flags = DEFAULT_FLAG;
+
+    dev_info_t dev_info;
+    dev_info.bus = I2C_BUS_3;
+    dev_info.addr = FAN_IO_CTL;
+    dev_info.offset = 0x00;
+    dev_info.flags = DEFAULT_FLAG;
+    
+    sprintf(fullpath, "%s%s", PREFIX_PATH, fan_path[local_id].speed);
+    rpm = dni_i2c_lock_read_attribute(&mux_info, fullpath);
```

Comment:
Just a suggestion -- this pattern happens often enough that you might consider making your dni_* routines take a format string and variable arguments to compose the "fullpath" buffer on behalf of the user. 
See the onlp_file_* APIs as an example.